backend/engine/manager/manager.py

```diff
@@ -93,6 +138,82 @@ def _register_plugins(self):
             EnginePlugin.objects.create(engine=self._engine, plugin=plugin, enabled=not settings.disabled)
             log.info("Registered plugin %s (enabled: %s)", plugin_name, not settings.disabled)
 
+    def set_instance_health(self):
```

Comment:
Except it doesn't necessarily set it to unhealthy. It only does that if the `_abnormal_shutdown` flag is true. Otherwise it leaves the instance health alone. I think this is a clearer name for where it is used (in the main execution loop).